From 31174c4105607cce969f23153313721d347172f7 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 23 May 2019 15:07:22 +0800
Subject: [PATCH] systemd: fix avc denials for systemd_modules_load_t

Fix:
type=AVC msg=audit(1557706838.234:115): avc:  denied  { search } for
pid=250 comm="systemd-modules" name="sys" dev="proc" ino=4026531854
scontext=system_u:system_r:systemd_modules_load_t
tcontext=system_u:object_r:sysctl_t tclass=dir permissive=1
type=AVC msg=audit(1557706838.234:115): avc:  denied  { search } for
pid=250 comm="systemd-modules" name="kernel" dev="proc" ino=10298
scontext=system_u:system_r:systemd_modules_load_t
tcontext=system_u:object_r:sysctl_kernel_t tclass=dir permissive=1
type=AVC msg=audit(1557706838.234:115): avc:  denied  { read } for
pid=250 comm="systemd-modules" name="osrelease" dev="proc" ino=10299
scontext=system_u:system_r:systemd_modules_load_t
tcontext=system_u:object_r:sysctl_kernel_t tclass=file permissive=1
type=AVC msg=audit(1557706838.234:115): avc:  denied  { open } for
pid=250 comm="systemd-modules" path="/proc/sys/kernel/osrelease"
dev="proc" ino=10299 scontext=system_u:system_r:systemd_modules_load_t
tcontext=system_u:object_r:sysctl_kernel_t tclass=file permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index e58d9b8..755eb1f 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -668,6 +668,21 @@ files_read_etc_files(systemd_modules_load_t)
 init_read_state(systemd_modules_load_t)
 init_search_run(systemd_modules_load_t)
 
+allow systemd_modules_load_t self:system module_load;
+kernel_load_module(systemd_modules_load_t)
+
+allow systemd_modules_load_t self:unix_dgram_socket { connect create getopt setopt };
+dontaudit systemd_modules_load_t self:capability sys_module;
+modutils_read_module_config(systemd_modules_load_t)
+modutils_read_module_deps(systemd_modules_load_t)
+kernel_read_system_state(systemd_modules_load_t)
+kernel_read_sysctl(systemd_modules_load_t)
+dev_read_sysfs(systemd_modules_load_t)
+kernel_dontaudit_search_kernel_sysctl(systemd_modules_load_t)
+kernel_dontaudit_read_kernel_sysctl(systemd_modules_load_t)
+logging_send_syslog_msg(systemd_modules_load_t)
+modutils_read_module_objects(systemd_modules_load_t)
+
 ########################################
 #
 # networkd local policy
-- 
2.7.4

